admin/acceptance.py

```diff
@@ -950,6 +951,101 @@ def eliot_output(message):
     sys.stdout.write(format % message)
     sys.stdout.flush()
 
+def capture_upstart(reactor, host, output_file):
+    """
+    SSH into given machine and capture relevant logs, writing them to
+    output file.
+
+    :param reactor: The reactor.
+    :param bytes host: Machine to SSH into.
+    :param file output_file: File to write to.
+    """
+    # TODO: check the format we need to use here (if we need it at all)
+    # can use it later. Using the whole filename instead of smtg like flocker*
+    # because I can't make it work inside the command. It has a funny way of
+    # parsing the commands. Suggestions are welcome :D
+    formatter = tail_formatter(output_file, host)
+    ran = run_ssh(
+        reactor=reactor,
+        host=host,
+        username='root',
+        command=[
+            b'tail',
+            b'/var/log/flocker/flocker-control.log',
+            b'/var/log/flocker/flocker-dataset-agent.log',
+            b'/var/log/flocker/flocker-container-agent.log',
+            b'/var/log/flocker/flocker-docker-plugin.log',
+            b'/var/log/upstart/docker.log',
+        ],
+        handle_stdout=formatter.handle_output_line,
+    )
+    ran.addErrback(write_failure, logger=None)
+    # Deliver a final empty line to process the last message
+    ran.addCallback(lambda ignored: formatter.handle_output_line(b""))
```

Comment:
You almost certainly want to return `ran` from here, so that anything that calls `capture_upstart` can wait for the results.